korgw/src/jvmMain/kotlin/com/soywiz/korgw/awt/AwtGameWindow.kt

```diff
@@ -378,14 +378,58 @@ class AwtGameWindow(val checkGl: Boolean) : GameWindow() {
                     MouseEvent.MOUSE_RELEASED -> com.soywiz.korev.MouseEvent.Type.UP
                     else -> com.soywiz.korev.MouseEvent.Type.MOVE
                 }
-                val id = 0
-                val x = e.x
-                val y = e.y
                 val button = MouseButton[e.button - 1]
                 val factor = frameScaleFactor
-                val sx = x * factor
-                val sy = y * factor
-                dispatchSimpleMouseEvent(ev, id, sx.toInt(), sy.toInt(), button, simulateClickOnUp = false)
+                val sx = e.x * factor
+                val sy = e.y * factor
+                val modifiers = e.modifiersEx
+                dispatchMouseEvent(
+                    type = ev,
+                    id = 0,
+                    x = sx.toInt(),
+                    y = sy.toInt(),
+                    button = button,
+                    buttons = 0,
+                    scrollDeltaX = 0.0,
+                    scrollDeltaY = 0.0,
+                    scrollDeltaZ = 0.0,
+                    isShiftDown = modifiers and MouseEvent.SHIFT_DOWN_MASK == MouseEvent.SHIFT_DOWN_MASK,
+                    isCtrlDown = modifiers and MouseEvent.CTRL_DOWN_MASK == MouseEvent.CTRL_DOWN_MASK,
+                    isAltDown = modifiers and MouseEvent.ALT_DOWN_MASK == MouseEvent.ALT_DOWN_MASK,
+                    isMetaDown = modifiers and MouseEvent.META_DOWN_MASK == MouseEvent.META_DOWN_MASK,
```

Comment:
To avoid MouseEvent.CONST repetition, you can use this function provide by kmem:

https://github.com/korlibs/korge-next/blob/67a34cad5cfa8aa02fe279cddec582162df91c8e/kmem/src/commonMain/kotlin/com/soywiz/kmem/Bits.kt#L149